homeassistant/components/geniushub/climate.py

```diff
@@ -0,0 +1,204 @@
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def min_temp(self):
+        """Return max valid temperature that can be set."""
+        return 4.0
+
+    @property
+    def max_temp(self):
+        """Return max valid temperature that can be set."""
+        return 28.0
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return OPERATION_LIST
+
+    @property
+    def is_on(self):
+        """Return true if the device is on."""
+        if self._mode == "off":
+            return False
+
+        return True
+
+    @property
+    def is_away_mode_on(self):
+        """Return true if away mode is on."""
+        if self._mode == "away":
+            return True
+
+        return False
+
+    @property
+    def current_operation(self):
+        """Return the current operation mode."""
+        return self.get_current_operation_mode(self._mode)
+
+    @staticmethod
+    def get_current_operation_mode(mode):
+        """Return the current operational mode."""
+        mode_map = {
+            'override': STATE_HEAT,
+            'footprint': STATE_ECO,
+            'timer': STATE_AUTO,
+        }
+        return mode_map.get(mode, STATE_IDLE)
+
+    def get_operation_mode(self, operation_mode):
+        """Coverts operation mode from Home Assistant to Genius Hub."""
+        # These needed to be mapped into HA modes:
+        # Off       => OFF      => STATE_IDLE   # Mode_Off: 1,
+        # Override  => HEAT     => STATE_HEAT # Mode_Boost: 16,
+        # Footprint => ECO      => STATE_ECO    # Mode_Footprint: 4,
+        # Timer     => AUTO     => STATE_AUTO   # Mode_Timer: 2,
+        # Away                      # Mode_Away: 8,
+        #
+        # OPERATION_LIST = [STATE_IDLE, STATE_HEAT, STATE_ECO, STATE_AUTO]
+        operation_mode_map = {
+            STATE_IDLE: {'mode': 'off', 'data': {'iMode': 1}},
+            STATE_HEAT: {'mode': 'override', 'data':
+                         {'iBoostTimeRemaining': 3600,
+                          'iMode': 16,
+                          'fBoostSP': self._target_temperature}},
+            STATE_ECO: {'mode': 'footprint', 'data': {'iMode': 4}},
+            STATE_AUTO: {'mode': 'timer', 'data': {'iMode': 2}}, }
+        return operation_mode_map.get(operation_mode,
+                                      {'mode': 'off', 'data': None})
+
+    async def async_set_operation_mode(self, operation_mode):
+        """Set new operation mode."""
+        data = self.get_operation_mode(operation_mode)
+        self._mode = data['mode']
+        if data['data'] is None:
+            _LOGGER.error("Unknown mode")
```

Comment:
Should we include what mode is unknown?

Reply:
We have been asked to set a mode. This is guarding about getting asked to set a mode that is not supported? What value does supporting mode is unknown do?

Reply:
I would include `operation_mode` in the log, so it's clear what mode is unknown.

Reply:
That's what the code is trying to do. get_operation_mode() always returns a result. If the ['data'] for the result is None we know that we have been requested with an unknown value, which we log.
This also has the effect of switch the heating off. We could just log the error and return.

I am open to suggestions to how the code could be improved is this is unclear.

Reply:
Please change the log message as below.
```py
_LOGGER.error("Unknown mode %s", operation_mode)
```

Reply:
Done.